File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.py
"""
"""
import sys

input = lambda: sys.stdin.readline().strip()

N = int(input())

i = 666
result = 0


def is_valid(i):
    repeat = 0
    while i:
        if i % 10 == 6:
            repeat += 1
            if repeat == 3:
                return True
        else:
            repeat = 0
        i //= 10
    return False


while True:
    if is_valid(i):
        result += 1
    if result == N:
        break
    i += 1

print(i)

File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;
    static StringBuilder sb = new StringBuilder();

    public static void main(String[] args) throws IOException {

        int N = Integer.parseInt(br.readLine());

        int index = 665;

        while (N != 0) {
            index += 1;
            if (isValid(index) == true) {
                N -= 1;
            }
        }

        System.out.println(index);

    } // end of main

    private static boolean isValid(int index) {
        int repeat = 0;
        while (index != 0) {
            if (index % 10 == 6) {
                repeat += 1;
                if (repeat == 3)
                    return true;
            } else {
                repeat = 0;
            }
            index /= 10;
        }
        return false;
    }

} // end of class